### DISCUSSION TITLE
How can I handle the error of useFetch?

### DISCUSSION THREAD

**Discussion by Vivi-Ornitier** [nuxt/nuxt#17337] [Questions] [answered]
I need to error handle the result of useFetch.

```
export const useAuth = async (body) => {
  const { data, error } = await useFetch("/v1/auth", {
    method: ”POST”,
    body,
    baseURL: useRuntimeConfig().baseURL,
    onResponseError({ response }) {
      throwError(new CustomError(response._data.resultCode, response.statusText));
    },
  });

  // Now I want to return an error to the caller.


}
```

Thanks.

**Comment by Flanker72** [comment]
You may simple use `throw` like this:
```
if (error.value) {
  throw error.value;
}
```

Also you can use a plain `$fetch` and handle its result as usual (try...catch).

**Reply by Vivi-Ornitier** [reply]
@Flanker72 
Thank you for your reply.

I want to know the Response body in case of an error.
However, error.value returns true.

Is there a way to refer to the Response body in case of an error?

When I tried to output error to the console, the contents were as follows.
```
{
    "__v_isShallow": false,
    "__v_isRef": true,
    "_rawValue": {
        "name": "FetchError"
    },
    "_value": {
        "name": "FetchError"
    }
}
```

This is not the actual response body, the actual response body has disappeared.

Thank,

**Reply by Flanker72** [reply]
Seems it may be done using `onResponseError` handler: https://github.com/unjs/ohmyfetch#onresponseerror-request-options-response-

**Comment by videmann** [comment, marked as answer]
I guess, you need to use const { data } = await useFetch([...]).catch((error) => error.data)

error info will be inside data
